pkg/server/home_workspaces.go

```diff
@@ -0,0 +1,661 @@
+	if rbacFound, searchRBACError := h.searchHomeWorkspaceRBACLocally(workspaceName, userName); searchRBACError != nil {
+		found = false
+		retryLater = false
+		retryAfterSeconds = 0
+		checkError = searchRBACError
+	} else if !rbacFound {
+		// Retry sooner than the creation delay, because it's probably a question of
+		// the local informer cache not being up-to-date, or the request having been sent
+		// to the wrong shard.
+		// Retrying quicly should be sufficient to fix it.
+		retryAfterSeconds = 1
+		retryLater = true
+	}
+	return found, retryLater, retryAfterSeconds, checkError
+}
+
+// tryToCreate tries to create the home or home bucket workspace, with an external client,
+// if it has not been found in the local informers.
+// Of course it can be that it has been created in the meantime (either concurrently on the current shard or
+// on another shard). We don't error in this case.
+// If it cannpot be created because the parent home bucket doesn't exist, create the parent first and
+// retry the creation of the current workspace later.
+// When creating a Home workspace, also create the related RBAC resources.
+func (h *homeWorkspaceHandler) tryToCreate(ctx context.Context, userName string, workspaceToCheck logicalcluster.Name, workspaceType tenancyv1alpha1.ClusterWorkspaceTypeName) (retryLater bool, retryAfterSeconds int, createError error) {
+	if h.mocks.tryToCreate != nil {
+		return h.mocks.tryToCreate(ctx, userName, workspaceToCheck, workspaceType)
+	}
+
+	// Try to create it in the parent workspace
+	parent, name := workspaceToCheck.Split()
+	createError = h.createClusterWorkspace(ctx, parent, &tenancyv1alpha1.ClusterWorkspace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: tenancyv1alpha1.ClusterWorkspaceSpec{
+			Type: tenancyv1alpha1.ClusterWorkspaceTypeReference{Path: "root", Name: workspaceType},
+		},
+	})
+
+	workspaceExists := createError == nil || kerrors.IsAlreadyExists(createError)
+	if workspaceExists {
+		if workspaceType == "Home" {
+			if err := h.createHomeRBAC(ctx, userName, workspaceToCheck); err != nil {
+				createError = err
+				return
+			}
+		}
+		// Retry sooner than the creation delay, because it's probably a question of
+		// letting the local informer cache being updated.
+		// Retrying quicly should be sufficient to it.
+		retryAfterSeconds = 1
+		retryLater = true
+		createError = nil
+		return
+	}
+
+	if retryAfterCreateSeconds, shouldRetry := kerrors.SuggestsClientDelay(createError); shouldRetry {
+		retryAfterSeconds = retryAfterCreateSeconds
+		retryLater = true
+		createError = nil
+		return
+	}
+
+	if kerrors.IsForbidden(createError) {
+		if parent == h.homePrefix {
+			// If the parent itself is the home root, which is created by default => return the error.
+			return
+		}
+
+		grandParent, parentName := parent.Split()
+		// Check in the grand parent if the parent itself exists
+		if parent, err2 := h.getClusterWorkspace(ctx, grandParent, parentName); err2 == nil {
```

Comment:
err

Reply:
Changed in commit 1f37994d81146bb38fbecd97319d1a5a24eba461